### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by egichuhis**
Hello, please review this PR.

Changes:

This pull request focuses on implementing user authentication using the devise gem. The key functionalities include user registration, login, email confirmation, and password reset.

Implementation Details

Authentication with Devise
Removed the current_user method in ApplicationController as it is now handled by Devise.
Integrated and configured the Devise gem for streamlined user authentication.

User Registration and Login
Users can now register with a new account using the provided registration form.
Implemented a secure login system where users authenticate using their email and password.

Password Security
Enhanced security by securely storing hashed passwords in the database.

Email Confirmation
Implemented the email confirmation feature, ensuring users must confirm their email addresses.

Password Reset
Users can initiate a secure password reset process to regain access to their accounts.

View Modifications
Aligned registration and login views with wireframes from the Sneak Peek, providing a cohesive user experience.

Thank you for your time and consideration.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # user.rb
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (TedLivist) [inline app/models/user.rb]: - [x] Users should be required to confirm their emails after signing up, before they can log in. Kindly add the `:confirmable` Devise module to fix this.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @egichuhis and @Rifat8080,

Good job so far!
✔️ Integration of Devise for authentication
✔️ Users can login with email and password
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by egichuhis** [comment]
Thanks @TedLivist for the reminder. Cheers!

**Review by gemmen29 — APPROVED** [review]
Hi @egichuhis, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**